### PR TITLE
feat: apply ignore pattern on context directory git source

### PR DESCRIPTION
**Pull request by TomChv**
Fixe based https://github.com/dagger/dagger/pull/8369 and @helderco feedbacks
 
**Without this fix**

```
dagger call -m github.com/wingyplus/dagger/sdk/php/runtime@cd3a3e724aa5a2ce259fe6b217f0efe5abaaabcb source-dir

_type: Directory
digest: sha256:4efcbbc3978405e30f078b41ff5acb6a0d1f6084906a42d3cf32d4377190a414
entries:
    - .changes
    - .changie.yaml
    - .gitattributes
    - .gitignore
    - .php-cs-fixer.dist.php
    - CHANGELOG.md
    - LICENSE
    - README.md
    - composer.json
    - composer.lock
    - dev
    - docker
    - docker-compose.yml
    - generated
    - phpcs.xml
    - phpunit.xml.dist
    - psalm.xml
    - runtime
    - scripts
    - src
    - tests
```

**With that fix**

```
dagger call -m github.com/wingyplus/dagger/sdk/php/runtime@cd3a3e724aa5a2ce259fe6b217f0efe5abaaabcb source-dir

_type: Directory
digest: sha256:87f768c76fbf1901d1b036461fc9207d82f7eccfc26d892540ef276e332aafbb
entries:
    - LICENSE
    - README.md
    - composer.json
    - composer.lock
    - generated
    - scripts
    - src
```

### CODE REVIEW

File: core/modulesource.go [hunk 1]
```diff
@@ -380,7 +380,30 @@ func (src *ModuleSource) LoadContext(ctx context.Context, dag *dagql.Server, pat
...
+		err = dag.Select(ctx, dag.Root(), &ignoredDir,
+			dagql.Selector{
+				Field: "directory",
+			},
+			dagql.Selector{
+				Field: "withDirectory",
+				Args: []dagql.NamedInput{
+					{Name: "path", Value: dagql.String("/")},
+					{Name: "directory", Value: dagql.NewID[*Directory](loadedDir.ID())},
+					{Name: "exclude", Value: dagql.ArrayInput[dagql.String](dagql.NewStringArray(ignore...))},
```
Comment (jedevc) [inline core/modulesource.go:398]: We should remove the `exclude`/`include` vars before the switch statement if we're doing it this way.

Comment (TomChv) [reply core/modulesource.go:398]: True

Comment (TomChv) [reply core/modulesource.go:395]: Done in [21f7519](https://github.com/dagger/dagger/pull/8430/commits/21f75197f329ce56cf4dfd3d4c8a86a25d71ea0a)

**Review by jedevc — APPROVED** [review]

**Comment by TomChv** [comment]
I just added the release note, gonna merge that now.

**Pull request merged**